.github/workflows/create-release-pr.yml

```diff
@@ -0,0 +1,127 @@
+                      "https://api.github.com/repos/${{ github.repository }}/releases")
+          tag=$(echo "$releases" | jq -r '.[0].tag_name')
+          echo "latest-tag=$tag" >> $GITHUB_OUTPUT
+
+      - name: Build Changelog
+        uses: mikepenz/release-changelog-builder-action@v3.7.2
+        id: build-changelog
+        env:
+          PREVIOUS_VERSION: ${{ steps.get-release.outputs.latest-tag }}
+        with:
+          fromTag: ${{ env.PREVIOUS_VERSION }}
+          toTag: ${{ steps.create-release-branch.outputs.release-branch }}
+          ignorePreReleases: true
+          failOnError: true
+          configurationJson: |
+            {
+                "categories": [
+                    {
+                        "title": "## New Features",
+                        "labels": [
+                            "New Feature"
+                        ]
+                    },
+                    {
+                        "title": "## Enhancement",
+                        "labels": [
+                            "Enhancement"
+                        ]
+                    },
+                    {
+                        "title": "## Bug Fixes",
+                        "labels": [
+                            "Bug-Fix"
+                        ]
+                    },
+                    {
+                        "title": "## Not Yet Enabled",
+                        "labels": [
+                            "Not-Yet-Enabled"
+                        ]
+                    }
+                ],
+                "ignore_labels": [
+                    "Skip-Release-Notes"
+                ],
+                "sort": {
+                    "order": "ASC",
+                    "on_property": "mergedAt"
+                },
+                "template": "#{{CHANGELOG}}",
+                "pr_template": "- #{{TITLE}} by #{{AUTHOR}} in ##{{NUMBER}}"
+            }
+
+      - name: Update Changelog
+        env:
+          CHANGELOG_CONTENT: ${{ steps.build-changelog.outputs.changelog }}
+          PREVIOUS_VERSION: ${{ steps.get-release.outputs.latest-tag }}
+        run: |
+          git branch --show-current
+          echo "$(tail -n +2 CHANGELOG.md)" > CHANGELOG.md
+          echo -e "# Changelog\n\n# ${RELEASE_VERSION}\n\n${CHANGELOG_CONTENT}**Full Changelog**: https://github.com/algorand/py-algorand-sdk/compare/${PREVIOUS_VERSION}...${RELEASE_VERSION}" | cat - CHANGELOG.md > temp && mv temp CHANGELOG.md
+
+      - name: Update version in setup.py
+        run: |
+          python3 scripts/bump_version.py ${RELEASE_VERSION:1}
+
+      - name: Commit Changes
+        uses: EndBug/add-and-commit@v9.1.3
+        with:
+          message: 'update CHANGELOG.md and setup.py version'
+
+      - name: Create Pull Request
+        env:
+          CHANGELOG_CONTENT: ${{ steps.build-changelog.outputs.changelog }}
+          PREVIOUS_VERSION: ${{ steps.get-release.outputs.latest-tag }}
+          GH_TOKEN: ${{ github.token }}
```

Comment:
Is this scoped to only work for release branches?

Reply:
Well, the source_ref to create the release_branch can come from anywhere including a hot_fix branch. 

The parameter: ```source_ref:
        description: 'Source branch, commit, or tag to create the release branch from'```

Reply:
Actually I removed the source_ref but I added a check to the "Create Release Branch if it does not exist" step which should ensure that we are on working on the release branch.